monitor/data_monitoring.py

```diff
@@ -156,25 +154,67 @@ def generate_report(self, days_back: Optional[int] = None, test_runs_amount: Opt
                 elementary_output_json_file:
             elementary_output_json_file.write(dumped_output_data)
 
-        webbrowser.open_new_tab('file://' + html_path)
+        if should_open_browser:
+            webbrowser.open_new_tab('file://' + html_path)
         self.execution_properties['report_end'] = True
         self.execution_properties['success'] = self.success
         return self.success, html_path
 
-    def send_report(self, elementary_html_path: str) -> bool:
-        if os.path.exists(elementary_html_path):
+    def send_report(self, html_path: str) -> bool:
+        if self.slack_client:
+            self.execution_properties['send_to_slack'] = True
             file_uploaded_successfully = self.slack_client.upload_file(
-                channel_name=self.slack_channel_name,
-                file_path=elementary_html_path,
+                channel_name=self.config.slack_channel_name,
+                file_path=html_path,
                 message=SlackMessageSchema(text="Elementary monitoring report")
             )
+            logger.info('Sent report to Slack.')
             if not file_uploaded_successfully:
+                logger.error('Failed to send report to Slack.')
                 self.success = False
-        else:
-            logger.error('Could not send Elementary monitoring report because it does not exist')
-            self.success = False
         return self.success
 
+    def upload_report(self, html_path: str):
+        if self.s3_client:
+            self._upload_report_to_s3(html_path)
+        if self.gcs_client:
+            self._upload_report_to_gcs(html_path)
+        return self.success
+
+    def _upload_report_to_s3(self, html_path: str):
```

Comment:
I think that s3 client should be a class with a method for upload and a method for updating buckets website.
data_monitoring shouldn't know so much about s3, and shouldn't be aware about its internals.